<?xml version="1.0"?>

<launch>
  
  <param name="world" value="curved_2lane"/>
  <rosparam file="$(find dbw_runner)/yaml/car.yaml" ns="car"/>
  <rosparam file="$(find dbw_runner)/yaml/track_2lane.yaml" ns="track"/>

  <include file="$(find dbw_mkz_gazebo)/launch/dbw_mkz_gazebo.launch" >
    <arg name="use_camera_control" value="false" />
    <arg name="world_name" value="$(find dbw_runner)/worlds/curved_2lane.world" />
    <arg name="sim_param_file" value="$(find dbw_runner)/yaml/curved_2lane.yaml" />
    <arg name="urdf_file" default="$(find dbw_runner)/urdf/mkz_bumpered.urdf.xacro"/>
  </include>
  
  <!--include file="$(find dbw_runner)/launch/lane_detection.launch" /-->
  <!--node ns="vehicle" pkg="dbw_mkz_vision_sim" type="path_following_node" name="path_following" /-->

  <include file="$(find dbw_mkz_can)/launch/dbw.launch" >
    <arg name="live" value="false" />
    <arg name="load_urdf" value="false" />
    <arg name="can_ns" value="can_bus_dbw" />
    <arg name="vehicle_ns" value="vehicle" />
  </include>


  <!--node name="main" pkg="dbw_runner" type="main.py" output="screen"/-->

  <arg name="dev" default="/dev/input/js0"/>
  <node pkg="joy" type="joy_node" name="joy_node" output="screen" >
    <param name="dev" value="$(arg dev)" />
    <param name="deadzone" value="0.01" />
    <param name="autorepeat_rate" value="50.0" />
  </node>

  <node name="image_view1" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/vehicle/front_camera/image_raw" />
  </node>

  <node name="image_view2" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/mkz/front_camera/image_raw" />
  </node>

  <node name="image_view3" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/vehicle/bird_camera/image_raw" />
  </node>

  <node name="image_view4" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/mkz/bird_camera/image_raw" />
  </node>
</launch>
